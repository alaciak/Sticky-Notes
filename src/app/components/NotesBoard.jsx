import React from 'react';
import Note from './Note.jsx';
import { connect } from 'react-redux';
import { Provider } from 'react-redux';
import store from '../Store.jsx';

class NotesBoard extends React.Component {

  render() {
    if(this.props.notesList.length === 0) {
      return null;
    }
    else {
      const noteList = this.props.notesList.map((el, index) => {
        return (
            <Provider store={ store } key={ index }>
              <Note noteId={ el.id }/>
            </Provider>
        );
      });
      return (
        <div className='row'> { noteList }</div>
        )
    }
  }
}


const mapStateToProps = (state) => {
  return {
    notesList: state.notesBoardReducer.notesList
  }
};

const mapDispatchToProps = (dispatch) => {
  return {}
  };

export default connect(mapStateToProps, mapDispatchToProps)(NotesBoard);
