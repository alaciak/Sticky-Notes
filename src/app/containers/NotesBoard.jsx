import React from 'react';
import PropTypes from 'prop-types';
import Note from './Note.jsx';
import { connect } from 'react-redux';
import { getNotes, moveNote } from '../actions/notesBoardActions';
import { Provider } from 'react-redux';
import store from '../Store.jsx';
import AlertContainer from 'react-alert';

const types = {
  note: 'note'
};

export class NotesBoard extends React.Component {
  alertOptions = {
    offset: 10,
    position: 'bottom left',
    theme: 'dark',
    transition: 'scale'
  };

  componentDidMount() {
    this.props.getNotes();
  }

  handleShowAlert = event => {
    this.msg.show('All Changes Saved', {
      time: 1000,
      type: 'success'
    });
  }

  render() {
    const { isOver, canDrop, connectDropTarget } = this.props;
    if(this.props.notesList.length === 0) {
      return null;
    } else if(this.props.loading) {
      return null;
    } else {
        const noteList = this.props.notesList.map((el, index) => {
          return (
              <Provider store={ store } key={ el.id }>
                <Note noteId={ el.id } note={ el } showAlert={ this.handleShowAlert} index={ index }/>
              </Provider>
          );
        });
        return (
          <div className='row'>
            { noteList }
            <AlertContainer ref={a => this.msg = a} {...this.alertOptions} />
          </div>
        );
      }
    }
  }


const mapStateToProps = (state) => {
  return {
    notesList: state.notesBoardReducer.notesList,
    loading: state.notesBoardReducer.loading
  }
};

const mapDispatchToProps = (dispatch) => {
  return {
    getNotes: () => {
      dispatch(getNotes())
    }
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(NotesBoard);
