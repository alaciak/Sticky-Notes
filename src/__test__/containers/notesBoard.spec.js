import React from 'react';
import { NotesBoard } from '../../app/containers/NotesBoard.jsx';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('NotesBoard', () => {

  it('should not render if the data are not fetched from the server', () => {

    const testNotesList = [{ id: 1, id: 2 }];

    const wrapper = shallow(
      <NotesBoard loading={ true } notesList= { testNotesList }/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();

  });

  it('should not render if the list from the server is empty', () => {

    const testNotesList = [];

    const wrapper = shallow(
      <NotesBoard loading={ false } notesList= { testNotesList }/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();

  });

  it('should render if the data are fetched from the server and the list is not empty', () => {

    const testNotesList = [{ id: 1, id: 2 }];

    const wrapper = shallow(
      <NotesBoard loading={ false } notesList= { testNotesList } />
    );

    expect(toJson(wrapper)).toMatchSnapshot();

  });

});
