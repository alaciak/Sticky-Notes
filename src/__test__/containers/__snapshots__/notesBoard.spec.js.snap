// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesBoard should not render if the data are not fetched from the server 1`] = `null`;

exports[`NotesBoard should not render if the list from the server is empty 1`] = `null`;

exports[`NotesBoard should render if the data are fetched from the server and the list is not empty 1`] = `
<div
  className="row"
>
   
  <Provider
    key="0"
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(Note)
      note={
        Object {
          "id": 2,
        }
      }
      noteId={2}
    />
  </Provider>
</div>
`;
